vignettes/GMAT-GMST.Rmd

```diff
@@ -0,0 +1,126 @@
+
+This vignette demonstrates how to set up the calculation between GMAT and GMST and how to plot results to observe differences over time. 
+
+(Note that Hector assumes that land air temperature and land surface temperature are equivalent.^[Kriegler, Elmar. "Imprecise Probability Analysis
+for Integrated Assessment of Climate Change." PhD thesis, University of Potsdam, 2005. https://www.pik-potsdam.de/members/edenh/theses/PhDKriegler.pdf])
+
+## Setup
+
+Let's load the packages we will need and run a scenario.
+
+``` {r hector.init}
+library(hector)
+library(ggplot2)
+
+inifile <- file.path(system.file('input', package = 'hector'), 'hector_rcp45.ini')
+core <- newcore(inifile)
+invisible(run(core))
+```
+
+
+## Find GMST
+
+GMST can be calculated from Hector outputs by computing a weighted sum of the land air and ocean surface temperatures using the following equation:
+
+$ GMST = (land air temperature * fractional land area) + (ocean surface temperature * (1 - fractional land area)) $
+
+In Hector, land air temperature is represented by `Tgav_land` (`LAND_AIR_TEMP()`) and ocean surface temperature is represented by `Tgav_ocean_ST` (`OCEAN_SURFACE_TEMP()`). These components stem from the variable `Tgav` (`GLOBAL_TEMP()`), which is how Hector stores GMAT.
+
+``` {r find.GMST}
+flnd = 0.29;     # fractional land area, a constant in Hector
+
+hector_output <- fetchvars(core, 1745:2300, vars = c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP(), GLOBAL_TEMP()))
+
+# Get the variables we want and corresponding years/value
+output <- subset(hector_output, variable %in% c(LAND_AIR_TEMP(), OCEAN_SURFACE_TEMP()))
+output <- subset(output, select = c(year, variable, value))
+
+# Reorder and clean up data
+output <- reshape(output, direction = "wide", idvar = "year", timevar = "variable")
+names(output) <- gsub("value.", "", names(output))
+
+# Conversion equation
+output$GMST <- with(output, (Tgav_land * flnd + Tgav_ocean_ST * (1-flnd)))
+global_surface_temp <- subset(output, select = c(year, GMST))
+
+# Organize final data, year and corresponding GMST
+names(global_surface_temp)[names(global_surface_temp) == "output$year"] <- "year"
+names(global_surface_temp)[names(global_surface_temp) == "GMST"] <- "value"
+
+variable <- rep("GMST", times = 556)
+variable <- as.data.frame(variable)
+global_surface_temp <- cbind(global_surface_temp, variable)
+
+# Rename Hector's variable for GMAT
+global_air_temp <- subset(hector_output, variable == "Tgav")
+global_air_temp <- subset(global_air_temp, select = c(variable, year, value))
+global_air_temp$variable <- "GMAT"
+
+```
+
+## Plot results
+
+The following plot shows GMST and GMAT over time. We see that in about 2000, GMAT begins to increase more quickly than GMST. By 2300, there is a significant gap between the two; GMAT is higher than GMST as GMAT is responding more quickly to anthropogenic forcing. This makes sense as GMST accounts for heating over the oceans which warm more slowly than the atmosphere. 
```

Comment:
Perhaps: "...GMST accounts for heating of the ocean surface, which warms more slowly than the atmosphere."